[BUG][C#][csharp-netcore] maxLength + enum = generated code doesn't build

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))


##### Description

When an object property features `maxLength` (and possibly other similar attributes?) alongside an `enum`, the generated code does not compile.
Now, I know this is kind of weird to have both enum values and maxLength defined but the spec doesn't prevent those for the reasons stated [here](https://github.com/OAI/OpenAPI-Specification/issues/2153#issuecomment-590190022).
Which means that the validators won't catch these, the generator will run fine, only at the final step of building a package the error would show up.

##### openapi-generator version

master @ e09409fbce

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.3
info:
  title: C# enum length repro
  version: v1
paths:
  '/hello':
    get:
      responses:
        '200':
          description: returns a greeting
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Greeting'

components:
  schemas:
    Greeting:
      type: object
      properties:
        reply:
          type: string
          maxLength: 20
          enum:
            - hi
            - howdy
```

##### Generation Details

See steps below.


##### Steps to reproduce

```
openapi-generator generate -i openapi.yaml -g csharp-netcore --library httpclient -o out/csharp_enum
cd out/csharp_enum 
dotnet build --configuration Release
```

produces:

```
(...)
out/csharp_enum/src/Org.OpenAPITools/Model/Greeting.cs(141,49): error CS1061: 'Greeting.ReplyEnum?' does not contain a definition for 'Length' and no accessible extension method 'Length' accepting a first argument of type 'Greeting.ReplyEnum?' could be found (are you missing a using directive or an assembly reference?)
(...)

Build FAILED.
```

##### Related issues/PRs

None that I could find.

##### Suggest a fix

No idea... 🙇 


@mandrean @frankyjuang @shibayan @Blackclaws @lucamazzanti @wing328 @prajon84

--- a/modules/openapi-generator/src/main/resources/csharp-netcore/modelGeneric.mustache
+++ b/modules/openapi-generator/src/main/resources/csharp-netcore/modelGeneric.mustache
@@ -300,6 +300,7 @@
             {{/parent}}
             {{#vars}}
             {{#hasValidation}}
+            {{^isEnum}}
             {{#maxLength}}
             // {{{name}}} ({{{dataType}}}) maxLength
             if(this.{{{name}}} != null && this.{{{name}}}.Length > {{maxLength}})
@@ -343,6 +344,7 @@
 
             {{/isByteArray}}
             {{/pattern}}
+            {{/isEnum}}
             {{/hasValidation}}
             {{/vars}}
             yield break;

